storage/src/nodestore/alloc.rs

```diff
@@ -768,6 +769,41 @@ impl<T, S: ReadableStorage> NodeStore<T, S> {
     }
 }
 
+fn read_bincode_varint_u64_le(reader: &mut impl Read) -> std::io::Result<u64> {
+    // See https://github.com/ava-labs/firewood/issues/1146 for full details.
+    // emulate this behavior: https://github.com/bincode-org/bincode/blob/c44b5e364e7084cdbabf9f94b63a3c7f32b8fb68/src/config/int.rs#L241-L258
+
+    const SINGLE_BYTE_MAX: u8 = 250;
+    const U16_BYTE: u8 = 251;
+    const U32_BYTE: u8 = 252;
+    const U64_BYTE: u8 = 253;
+
+    match reader.read_byte()? {
+        byte @ 0..=SINGLE_BYTE_MAX => Ok(u64::from(byte)),
```

Comment:
TIL: `@` is valid syntax in rust

Reply:
this also works anywhere there is a pattern.

```rust
match PartialCmp::partial_cmp(&left, &right) {
    Some(cmp @ (Ordering::Less | Ordering::Greater)) => todo!(),
    var @ (Some(Ordering::Equal) | None) => return var,
}
```